python/hail/methods/io.py

```diff
@@ -0,0 +1,219 @@
+from hail.api2 import MatrixTable, Table
+from hail.genetics import GenomeReference
+from hail.utils.java import handle_py4j, Env
+from hail.typecheck import *
+
+@handle_py4j
+@typecheck(path=strlike,
+           reference_genome=nullable(GenomeReference))
+def import_interval_list(path, reference_genome=None):
+    """Import an interval list file in the GATK standard format.
+
+    >>> intervals = KeyTable.import_interval_list('data/capture_intervals.txt')
+
+    The File Format
+    ---------------
+
+    Hail expects an interval file to contain either three or five fields per
+    line in the following formats:
+
+    - ``contig:start-end``
+    - ``contig  start  end`` (tab-separated)
+    - ``contig  start  end  direction  target`` (tab-separated)
+
+    A file in either of the first two formats produces a key table with one
+    column:
+
+     - **interval** (*Interval*), key column
+
+    A file in the third format (with a "target" column) produces a key with two
+    columns:
+
+     - **interval** (*Interval*), key column
+     - **target** (*String*)
+
+    Note
+    ----
+    ``start`` and ``end`` match positions inclusively, e.g.
+    ``start <= position <= end``. :meth:`representation.Interval.parse`
+    is exclusive of the end position.
+
+    Note
+    ----
+    Hail uses the following ordering for contigs: 1-22 sorted numerically, then
+    X, Y, MT, then alphabetically for any contig not matching the standard human
+    chromosomes.
+
+    Warning
+    -------
+    The interval parser for these files does not support the full range of
+    formats supported by the python parser
+    :meth:`representation.Interval.parse`. 'k', 'm', 'start', and 'end' are all
+    invalid motifs in the ``contig:start-end`` format here.
+
+    Parameters
+    ----------
+    filename : :obj:`str`
+        Path to file.
+
+    reference_genome : :class:`.GenomeReference`
+        Reference genome to use. Default is
+        :class:`.HailContext.default_reference`.
+
+    Returns
+    -------
+    :class:`.Table`
+        Interval-keyed table.
+    """
+
+    rg = reference_genome if reference_genome else Env.hc().default_reference
+    t = Env.hail().table.Table.importIntervalList(Env.hc()._jhc, path, rg._jrep)
+    return Table(t)
+
+@handle_py4j
+@typecheck(path=strlike,
+           reference_genome=nullable(GenomeReference))
+def import_bed(path, reference_genome=None):
+    """Import a UCSC .bed file as a key table.
+
+    Examples
+    --------
+
+    Add the variant annotation ``va.cnvRegion: Boolean`` indicating inclusion in
+    at least one interval of the three-column BED file `file1.bed`:
+
+    >>> bed = methods.import_bed('data/file1.bed')
+    >>> vds_result = vds.annotate_rows(cnvRegion = bed[vds.v])
```

Comment:
this doesn't work right now.

Reply:
as in wrong syntax? Or I should be using something else?

Reply:
the `bed` is keyed by `Interval[Locus]`, which can't be joined with `vds.v` (`Variant`). I think we still don't have a great story for how this will look. Maybe it'll look something like this and do interval joins automatically behind the scenes, or maybe we'll require a separate method.